### PR TITLE
add template for pipenv, update install routine

### PR DESCRIPTION
**Pull request by LeoIV**
There is an issue with `autohooks` being installed in `pipenv` environments. The hook is installed correctly but it cannot be run because it starts a `python3` environment outside of the pipenv. However, `autohooks` cannot be imported in this case as it is only known in the virtual environment.

This PR checks if a Pipenv is present in the current project. If this is the case, it uses a different template that starts the hook with `pipenv run python3`.

**Checklist**:

- [ ] Tests
- [ ] [CHANGELOG](https://github.com/greenbone/autohooks/blob/master/CHANGELOG.md) Entry
- [x] Documentation

### CODE REVIEW

File: autohooks/install.py [hunk 1]
```diff
@@ -32,10 +33,15 @@ def get_pre_commit_hook_path():
 
 
 def get_pre_commit_hook_template_path():
-    setup_dir_path = get_autohooks_directory_path()
-    return setup_dir_path / 'precommit' / 'template'
+    # check if in pipenv
+    ret_code = os.system("pipenv graph")
```
Comment (bjoernricks) [inline autohooks/install.py]: Please always use subprocess instead of os.system

**Comment by LeoIV** [comment]
I didn't read the `README.md` careful enough. I think it would be useful for many applications if the hooks are installed and enabled automatically with the project checkout.

I don't see how this can be done with a command line switch as this requires the user to set this option explicitly.

**Comment by bjoernricks** [comment]
I really would like to support this use case! Maybe some we could use a setting in the pyproject.toml file instead?

**Comment by LeoIV** [comment]
Perfect, I will look into this soon 🙂

**Comment by LeoIV** [comment]
@bjoernricks I added the option which can be enabled in the `pyproject.toml` (details are in the `README.md`)

**Comment by LeoIV** [comment]
Done

**Comment by LeoIV** [comment]
Nah, you're right. The hook is installed just like the other but is run automatically.

File: autohooks/install.py [hunk 2]
```diff
@@ -17,13 +17,17 @@
...
 
+from autohooks.config import load_config_from_pyproject_toml
 from autohooks.utils import (
     get_git_hook_directory_path,
     get_autohooks_directory_path,
-)
+    get_pyproject_toml_path)
+
+pyproject_toml = get_pyproject_toml_path()
+config = load_config_from_pyproject_toml(pyproject_toml)
```
Comment (bjoernricks) [inline autohooks/install.py]: Could you move the pyproject_toml and config variables down into the get_pre_commit_hook_template_path function? Module variables can be imported from other modules and both variables are only used in the get_pre_commit_hook_template_path function.

**Comment by LeoIV** [comment]
I implemented your change requests. Please let me know your decision regarding the naming and the default behavior.

**Comment by bjoernricks** [comment]
> By the way: I wrote a first version of an autopep8 plugin. How do you think about adding a link to it to the README.md?

Nice. Would be fine. Just double check your sources for old black references.

Personally I am thinking about more generic plugins for linting and formatting where theactual formatter and linter can be set via the config.

**Comment by bjoernricks** [comment]
> I implemented your change requests. Please let me know your decision regarding the naming and the default behavior.

I've thought a lot about this feature at the weekend. I would like to use a mode setting instead of a single boolean setting to be more extendable for the future: `mode = "pipenv"`.

Additionally I would like to convert the template hook file into a real template instead of providing two files. I've already laid out the groundwork for this change yesterday.

**Comment by LeoIV** [comment]
That sound reasonable. How would you like to proceed with this PR then?

**Comment by bjoernricks** [comment]
You could rebase your commit on top of my changes.

**Comment by bjoernricks** [comment]
Just added a PR for the mode config api in https://github.com/greenbone/autohooks/pull/24

**Comment by LeoIV** [comment]
Thanks for your PR and your willingness to support the feature :-) 

I think the changes in this PR are contained in your PR, so I will close this.

**Comment by bjoernricks** [comment]
> Thanks for your PR and your willingness to support the feature :-)

Really would love to get you proposed feature include. Just a matter of time and help.

> I think the changes in this PR are contained in your PR, so I will close this.

It's still missing the actual installation of the pre-commit hook from the template.

**Comment by bjoernricks** [comment]
@LeoIV  just wanted to let you know i've finished my work on #24 and it will get merged into master next week. With #24 it will possible to auto activate the pipenv env when running the git hook.

Btw. would love to get a PR to mention your pep8 plugin at the README!

**Pull request closed without merging**